doc/qbk/07_concepts/BuffersGenerator.qbk

```diff
@@ -0,0 +1,128 @@
+__message_generator__ which provides a type-erased interface for a variety of
+concrete http message types.
+
+Overloads of `write` and `async_write` operations are provided as free
+functions. These operations will consume the output of a [*BuffersGenerator]
+and process the data by writing them to a __SyncWriteStream__ or
+__AsyncWriteStream__ respectively.
+
+[heading Associated Types]
+
+* [link beast.ref.boost__beast__core__is_buffers_generator `is_buffers_generator`]
+* __ConstBufferSequence__
+
+[heading Requirements]
+
+[warning
+    These requirements may undergo non-backward compatible
+    changes in subsequent versions.
+]
+
+In this table:
+
+* `G` denotes a type meeting the requirements of [*BuffersGenerator].
+* `g` denotes a value of type `G`.
+* `n` is a value of type `std::size_t`.
+* `ec` is a value of type [link beast.ref.boost__beast__error_code `error_code&`].
+
+[table Valid expressions
+[[Expression] [Type] [Semantics, Pre/Post-conditions]]
+[
+    [`G::const_buffers_type`]
+    [`implementation-defined`]
+    [
+        A type which meets the requirements of __ConstBufferSequence__.
+        This is the type of buffer returned by `g.prepare(ec)`.
+    ]
+][
+    [`g.prepare(ec)`]
+    [`G::const_buffers_type`]
+    [
+        Called to ask the generator to produce buffers containing data for processing.
+
+        [*Note:] any buffers obtained by previous calls to `prepare` are
+        invalidated.
+        
+        The (maximum) capacity of the buffer returned is defined by the
+        generator implementation.
+
+        Whether additional buffers are generated when unconsumed data is
+        already available at the time of the call is defined by the implementation.
+
+        The returned value is the __ConstBufferSequence__ of type `b`
+        representing unconsumed data.
+
+        If the total size of the buffers in the returned sequence is `0`, the
+        generator is considered to have completed.
+
+        The function will ensure that `!ec` is `true` if there was
+        no error or set to the appropriate error code if there was one. 
+
+        The result of invoking `prepare` again after completion or encountered
+        error(s) is defined by the generator implementation. It can not be
+        assumed to be meaningful or safe to do so, in general.
+    ]
+][
+    [`g.consume(n)`]
+    [`void`]
+    [
+        This function is called to signal that the consumer (caller) of the
+        generator has processed part of the data returned by the previous call
+        to `prepare`.
+
+        The value of `n` is allowed to exceed the number of bytes returned from
+        the last call to `prepare`, in which case the call to `consume` shall
+        behave as if `n` was equal to that number.
+
+        Any previously generated data that has not yet been `consume`d will
+        be returned from subsequent calls to `prepare`.
+    ]
+][
+    [`is_buffers_generator<G>`]
```

Comment:
This could be a link I suppose. Do I link them in the other concepts?

Reply:
Other concepts don't reflexively include their own trait. I think I got that from an Asio concept. Keep it or drop it?

Reply:
I'm not sure... it does look nice in the list